include/cask/scheduler/SingleThreadScheduler.hpp

```diff
@@ -10,67 +10,83 @@
 #include <condition_variable>
 #include <map>
 #include <mutex>
+#include <optional>
 #include <thread>
 #include <queue>
 #include <vector>
 
 #include "../Scheduler.hpp"
-#include "SpinLock.hpp"
 
 namespace cask::scheduler {
+
+
     
 /**
- * The single thread scheduler only utilizes a single thread for processing submitted
- * work. It uses simpler lock-free protections where possible to protect its internal
- * structures. The result is a scheduler that is more optimized for applications where
- * the need for computational span across cores is not as strong. The benefit is greater
- * cache coherency and less time interacting with the OS for things like mutexes resulting
- * in faster handling of asynchrounous boundaries. This comes at the expensive of some amount
- * of fairness (because of spinlocks) and the inability to spread load across multiple cores.
+ * The single thread scheduler only utilizes a single thread for processing
+ * submitted work.
  */
 class SingleThreadScheduler final : public Scheduler, public std::enable_shared_from_this<SingleThreadScheduler> {
 public:
     /**
      * Construct a single threaded scheduler.
      */
-    explicit SingleThreadScheduler(int priority = 0);
+    explicit SingleThreadScheduler(
```

Comment:
Probably don't need 'explicit' here since it's no longer a single parameter constructor.